src/test/java/seedu/address/model/medicine/PricePerUnitTest.java

```diff
@@ -0,0 +1,39 @@
+package seedu.address.model.medicine;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import seedu.address.testutil.Assert;
+
+public class PricePerUnitTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        Assert.assertThrows(NullPointerException.class, () -> new PricePerUnit(null));
+    }
+
+    @Test
+    public void constructor_invalidPricePerUnit_throwsIllegalArgumentException() {
+        String invalidPricePerUnit = "";
+        Assert.assertThrows(IllegalArgumentException.class, () -> new PricePerUnit(invalidPricePerUnit));
+    }
+
+    @Test
+    public void isValidPricePerUnit() {
+        // null phone number
```

Comment:
Should the comment be null price per unit instead?

Reply:
Same for the other two comments below

Reply:
changed thanks!